amundsen_application/base/base_announcement_client.py

```diff
@@ -31,20 +32,16 @@ def _create_error_response(message: str) -> Response:
             return make_response(payload, HTTPStatus.INTERNAL_SERVER_ERROR)
 
         try:
-            try:
-                announcements = self.get_posts()
-            except Exception as e:
-                message = 'Encountered exception getting posts: ' + str(e)
-                return _create_error_response(message)
-
-            # validate the returned object
-            data, errors = AnnouncementsSchema().dump(announcements)
-            if not errors:
-                payload = jsonify({'posts': data.get('posts'), 'msg': 'Success'})
-                return make_response(payload, HTTPStatus.OK)
-            else:
-                message = 'Announcement data dump returned errors: ' + str(errors)
-                return _create_error_response(message)
+            announcements = self.get_posts()
         except Exception as e:
-            message = 'Encountered exception: ' + str(e)
+            message = 'Encountered exception getting posts: ' + str(e)
+            return _create_error_response(message)
+
+        try:
+            data = AnnouncementsSchema().dump(announcements)
```

Comment:
this dump-then-load pattern is ugly, but I think it may be the best way to keep the functionality unchanged. See https://marshmallow.readthedocs.io/en/stable/upgrading.html#validation-does-not-occur-on-serialization. 

In general, marshmallow doesn't seem to understand the idea of validating an app object. `load` or `validate` only work on dictionaries. This seems odd to me. If anyone knows more than me, please LMK.